### DISCUSSION TITLE
PackageReference PrivateAssets copies files to referenced project

### DISCUSSION THREAD

**Discussion by amaitland** [NuGet/Home#10631] [General]
Hi,

I was hoping on some clarification of the expected behaveiour when using `<PrivateAssets/>`.

> These assets will be consumed but won't flow to the parent project

According to https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets

There's no definition for `parent project` so I'm wondering if I'm misunderstanding the intended behaviour. 

I have a class library that includes
```xml
  <PropertyGroup>
    <TargetFrameworks>net462;netcoreapp3.1;net5.0-windows</TargetFrameworks>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="CefSharp.Common.NETCore" Version="88.2.90">
      <IncludeAssets>all</IncludeAssets>
      <PrivateAssets>build;buildTransitive;runtime;native;</PrivateAssets>
    </PackageReference>
  </ItemGroup>
```

An WinExe project targeting `netcoreapp3.1` that includes a `ProjectReference` to the class library.

Only specifying `all` appears to make a difference, I then get a build error. If I try `runtime` I'd also expect a build error. As the referenced class is missing. Every combination of `<PrivateAssets/>` I've tried that isn't `all` appears to do nothing.

Is `PrivateAssets` supposed to limit which files are copied to a referencing project? Have I totally missed the point?

Hope this is clear, appreciate any assistance.

```
Microsoft Visual Studio Community 2019
Version 16.8.6
dotnet sdk 5.0.103
```

**Comment by marcin-krystianc** [comment]
`Runtime` affects assets at execution time and `Compile` is for assets at compilation time. Did you try `<PrivateAssets>Compile</PrivateAssets>` ?

**Comment by maiosoft** [comment]
Hi Marcin, I tried as you suggested, but nothing changes when the project type is 'WinExe'.
Instead, when the project type is "Library" everything is fine.

Hoping not to bore you with my details, I am trying to achieve this behaviour:

I have created a nuget package containing an assembly of custom WinForms controls; this package is needed to have the items in the toolbox in projects that use these controls.
For other reasons, I already have this assembly in the build output folder and would like to use the latter and not the one that comes from the package.
They are totally compatible, except that the one in the bin folder contains runtime bugfixes, which are not needed at design time.

The point is that I only need the package to populate the toolbox (and the "automatically populate toolbox" flag is no good because it works if you reference the design and not the assembly) and I would like to avoid generating a new package for every bugfix that is not related to the design.

With PrivateAssets="runtime" I got the expected behaviour when compiling the "Library" projects, but no success with "WinExe".

I don't understand why...

**Reply by marcin-krystianc** [reply]
Hi @maiosoft,

I think I understand your use case, but It is hard to give an answer without actually seeing the project files. I can only suggest looking at the `project.assets.json` file for the `WinExe` project. It should give you a hint where is a particular assembly for given asset type (`compile` vs `runtime`) coming from. You can also try using [MSBuild Binary Log Viewer](https://msbuildlog.com/ ) to analyse the build process and what copy operations are being executed.
